dbsim/db_server_service.cpp

```diff
@@ -150,3 +150,16 @@ void db::server_service::debug_sync(const char*)
 {
 
 }
+
+int db::server_service::do_crypt(void**                ctx         WSREP_UNUSED,
+                                 wsrep::const_buffer&  key         WSREP_UNUSED,
+                                 unsigned int          key_version WSREP_UNUSED,
+                                 const char            (*iv)[32]   WSREP_UNUSED,
+                                 wsrep::const_buffer&  input       WSREP_UNUSED,
+                                 void*                 output      WSREP_UNUSED,
+                                 bool                  encrypt     WSREP_UNUSED,
+                                 bool                  last        WSREP_UNUSED)
+{
+    return 0;
```

Comment:
Returning 0 here may be confusing and lead to endless loop. I'd suggest either
 * to do a simple `memcpy()` from input to output and return the number of bytes copied
or
* return -ENOSYS or something like that

Reply:
returning -ENOSYS;

Reply:
I would avoid using system error codes unless actually interfacing with system calls. The error messages generated from system errors tend to be confusing and usually require switch case error handling if more fine grained error handling is needed.

I'm not fan of returning positive values for sizes and negative values for error codes either, it tends to make error handling code overly complex.

At this level it would be acceptable to throw an exception in case of error and translate it to error code for wsrep api in wsrep_provider_v25.cpp. Errors from these calls should be rare I guess.